cloudigrade/util/aws.py

```diff
@@ -10,6 +10,38 @@
 logger = logging.getLogger(__name__)
 
 
+class InstanceState(enum.Enum):
+    """
+    Enumeration of EC2 Instance state codes.
+
+    See also:
+        https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_InstanceState.html
+
+    """
+
+    pending = 0
+    running = 16
+    shutting_down = 32
+    terminated = 48
+    stopping = 64
+    stopped = 80
+
+    @classmethod
+    def is_running(cls, code):
+        """
+        Check if the given code is effectively a running state.
+
+        Args:
+            code (int): The code from an EC2 Instance state.
+
+        Returns:
+            bool: True if we consider the instance to be running else False.
+
+        """
+        return code in (cls.running.value,
+                        cls.shutting_down.value)
```

Comment:
Should we consider shutting down as a running instance? There will be no event written to CloudTrail when it does shut down, as such I think we may have to treat a shutting down state as already turned off.

Reply:
Hmm… I haven't actually tried this myself, but I had (maybe wrongly) expected CT to send an event when the shutdown was complete. Where did you find your info?

Reply:
Testing. CloudTrails doesn't log states of machines. It logs actions, so there is a log made when you as a user make a call to `stopInstance()`, which is when it'll go into the `shutting_down` state.

Reply:
Good to know! I'll pull this out.